crates/optimism/flashblocks/src/service.rs

```diff
@@ -36,12 +37,62 @@ pub struct FlashBlockService<
 > {
     rx: S,
     current: Option<PendingBlock<N>>,
-    blocks: Vec<FlashBlock>,
+    blocks: FlashBlockSequence,
     evm_config: EvmConfig,
     provider: Provider,
     canon_receiver: CanonStateNotifications<N>,
 }
 
+/// Simple wrapper around an ordered B-tree to keep track of a sequence of flashblocks by index
+#[derive(Debug)]
+struct FlashBlockSequence {
+    inner: BTreeMap<u64, FlashBlock>,
+}
```

Comment:
this type is out of order, this should be moved afer the impl blocks of FlashBlockService